itest/rust/src/object_tests/validate_property_test.rs

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use godot::builtin::{Array, Dictionary, GString, StringName};
+use godot::classes::IObject;
+use godot::global::PropertyUsageFlags;
+use godot::meta::PropertyInfo;
+use godot::obj::NewAlloc;
+use godot::register::{godot_api, GodotClass};
+use godot::test::itest;
+
+#[derive(GodotClass)]
+#[class(base = Object, init)]
+pub struct ValidatePropertyTest {
+    #[var(
+        hint = NONE,
+        hint_string = "initial"
+    )]
+    #[export]
+    my_var: i64,
+}
+
+#[godot_api]
+impl IObject for ValidatePropertyTest {
+    fn validate_property(&self, property: &mut PropertyInfo) {
+        if property.property_name.to_string() == "my_var" {
+            property.usage = PropertyUsageFlags::NO_EDITOR;
+            property.property_name = StringName::from("SuperNewTestPropertyName");
+            property.hint_info.hint_string = GString::from("SomePropertyHint");
+
+            // Makes no sense but allows to check if given ClassName can be properly moved to GDExtensionPropertyInfo.
+            property.class_name = <ValidatePropertyTest as godot::obj::GodotClass>::class_name();
+        }
+    }
+}
+
+#[itest]
+fn validate_property_test() {
+    let obj = ValidatePropertyTest::new_alloc();
+    let properties: Array<Dictionary> = obj.get_property_list();
+
+    for property in properties.iter_shared() {
+        if !property
+            .get("name")
+            .is_some_and(|v| v.to_string() == "SuperNewTestPropertyName")
+        {
+            continue;
+        }
+
+        let Some(hint_string) = property.get("hint_string").map(|v| v.to::<GString>()) else {
+            continue;
+        };
+        assert_eq!(hint_string, GString::from("SomePropertyHint"));
+
+        let Some(class) = property.get("class_name").map(|v| v.to::<StringName>()) else {
+            continue;
+        };
+        assert_eq!(class, StringName::from("ValidatePropertyTest"));
+
+        let Some(usage) = property.get("usage").map(|v| v.to::<PropertyUsageFlags>()) else {
+            continue;
+        };
+        assert_eq!(usage, PropertyUsageFlags::NO_EDITOR);
+
+        obj.free();
+        return;
+    }
```

Comment:
The individual `continue` statements define an order, i.e. those tested later are never reached if earlier ones fall into the `continue` branch. Is this intended?

Also, you currently skip each test when a key like `"usage"` is absent -- shouldn't you verify that it's present and has a given value? `Dictionary::at()` may help here as well.

Reply:
Changed to explicit panic in case if given value can't be retrieved (should be easier to spot during run).